python/ray/monitor.py

```diff
@@ -106,18 +108,19 @@ def __init__(self,
         self.load_metrics = LoadMetrics(local_ip=head_node_ip)
         self.last_avail_resources = None
         self.event_summarizer = EventSummarizer()
-        if autoscaling_config:
+        self.prefix_cluster_info = prefix_cluster_info
+        self.autoscaling_config = autoscaling_config
+        self.autoscaler = None
+
+        logger.info("Monitor: Started")
+
+    def _initialize_autoscaler(self):
+        if self.autoscaling_config:
             self.autoscaler = StandardAutoscaler(
-                autoscaling_config,
+                self.autoscaling_config,
                 self.load_metrics,
-                prefix_cluster_info=prefix_cluster_info,
+                prefix_cluster_info=self.prefix_cluster_info,
                 event_summarizer=self.event_summarizer)
```

Comment:
@wuisawesome I opted to separate out the autoscaler initialization from the constructor and move the failure handling into `Monitor.run()` in an effort to make it a bit cleaner.